packages/create-start/src/directory.ts

```diff
@@ -0,0 +1,81 @@
+import path from 'node:path'
+import fs from 'node:fs/promises'
+import { InvalidArgumentError, createOption } from '@commander-js/extra-typings'
+import { input } from '@inquirer/prompts'
+import {
+  checkFolderExists,
+  checkFolderIsEmpty,
+} from './utils/helpers/base-utils'
+
+export const getAbsolutePath = (relativePath: string) => {
+  return path.resolve(process.cwd(), relativePath)
+}
+
+const doesPathExist = async (absolutePath: string) => {
+  try {
+    await fs.access(absolutePath)
+    return true
+  } catch {
+    return false
+  }
+}
+
+const isFolderEmpty = async (absolutePath: string) => {
+  try {
+    const files = await fs.readdir(absolutePath)
+    return files.length === 0
+  } catch {
+    return false
+  }
+}
+
+const DEFAULT_NAME = 'my-tanstack-start-app'
+
+const generateDefaultName = async () => {
+  // Generate a unique default name e.g. my-tanstack-start-app,
+  // my-tanstack-start-app-1, my-tanstack-start-app-2 etc
+
+  let folderName = DEFAULT_NAME
+  let absolutePath = getAbsolutePath(folderName)
+  let pathExists = await doesPathExist(absolutePath)
+  let counter = 1
+  while (pathExists) {
+    folderName = `${DEFAULT_NAME}-${counter}`
+    absolutePath = getAbsolutePath(folderName)
+    pathExists = await doesPathExist(absolutePath)
+    counter++
+  }
+  return `./${folderName}`
+}
+
+export const newProjectDirectoryCliOption = createOption(
```

Comment:
can we move the logic of `newProjectDirectoryCliOption` and `newProjectDirectoryPrompt` into one function to avoid repetition?

Reply:
I'm not sure what you mean